### PR TITLE
Minor improvements and code refactoring

### PR DESCRIPTION
**Pull request by andrey-churkin**
### Changes

- The algorithm's backends for Bias Correction and Fast Bias Correction were refactored. The two modules were created:
    - `nncf.onnx.graph.node_utils`: Contains methods to get information from `NNCFNode`.
    - `nncf.onnx.graph.transformations.commands_creation`: Contains methods for commands creation.

- Minor refactoring of the algorithm's implementation was done. The redundant methods were removed.

### Reason for changes

Code duplication. The same methods are required for common Accuracy Aware algorithm implementation.

### Related tickets

N/A

### Tests

N/A

### CODE REVIEW

**Comment by andrey-churkin** [comment]
Guys, Could you please look at this stuff? If you don't have critical concerns I will update the BC (honest) algorithm as well.

File: nncf/quantization/algorithms/bias_correction/algorithm.py [hunk 1]
```diff
@@ -136,14 +135,16 @@ def _apply(self,
...
+            node for node in nncf_graph.topological_sort() if self._backend_entity.is_node_with_bias(node)
+        ]
+        subgraphs_data = [
+            self._get_subgraph_data_for_node(node, nncf_graph) for node in nodes_with_bias
+        ]
 
-            if not self._backend_entity.is_node_with_bias(node):
-                nncf_logger.debug(f'Skipping node {node_name} because there is no bias')
-                continue
+        for pos, (node, subgraph_data) in enumerate(zip(nodes_with_bias, subgraphs_data)):
```
Comment (KodiaqQ) [inline nncf/quantization/algorithms/bias_correction/algorithm.py:146]: What is the `pos`? Position?

File: nncf/quantization/algorithms/fast_bias_correction/algorithm.py [hunk 2]
```diff
@@ -148,24 +147,23 @@ def _apply(self,
...
-                bias_correction_command = self._backend_entity.bias_correction_command(target_point,
-                                                                                       updated_bias)
-                transformation_layout.register(bias_correction_command)
+                node_and_new_bias_value.append((node, updated_bias))
             else:
                 nncf_logger.debug(f'{node_name} bias skipped by threshold')
 
-        quantized_model = model_transformer.transform(transformation_layout)
-        return quantized_model
+        # Create commands of bias correction and apply them to the model.
```
Comment (KodiaqQ) [inline nncf/quantization/algorithms/fast_bias_correction/algorithm.py:159]: What is the difference/profit between the previous code?

File: nncf/quantization/algorithms/fast_bias_correction/backend.py [hunk 3]
```diff
@@ -132,26 +125,15 @@ def create_blob(shape: Tuple[int], data: List[float]) -> np.ndarray:
 
     @staticmethod
     @abstractmethod
-    def get_bias_value(model: TModel, node: NNCFNode) -> np.ndarray:
+    def get_bias_value(node: NNCFNode, model: TModel) -> np.ndarray:
```
Comment (KodiaqQ) [inline nncf/quantization/algorithms/fast_bias_correction/backend.py:128]: Why it was changed?

File: nncf/onnx/graph/transformations/commands_creation.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+"""
```
Comment (KodiaqQ) [inline nncf/onnx/graph/transformations/commands_creation.py:1]: Maybe it's better to call this file `commands_factory`?

File: nncf/quantization/algorithms/fast_bias_correction/algorithm.py (hunk 2 above)
Comment (andrey-churkin) [reply nncf/quantization/algorithms/fast_bias_correction/algorithm.py:159]: The main purpose of this change is to separate the command creation logic and the logic of the new bias calculation. It makes the code a little bit easy.

File: nncf/onnx/graph/transformations/commands_creation.py (hunk 4 above)
Comment (andrey-churkin) [reply nncf/onnx/graph/transformations/commands_creation.py:1]: I think we may use the command_factory name only if the `create_bias_correction_command()` method is the factory method.

File: nncf/onnx/graph/node_utils.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+from nncf.onnx.graph.metatypes.onnx_metatypes import LAYERS_WITH_BIAS_METATYPES
+from nncf.onnx.graph.onnx_graph import ONNXGraph
+from nncf.onnx.graph.metatypes.onnx_metatypes import ONNX_OPERATION_METATYPES
+from nncf.onnx.graph.metatypes.onnx_metatypes import ONNXIdentityMetatype
+
+
+def is_node_with_bias(node: NNCFNode) -> bool:
+    """
+    :param node:
+    :return:
```
Comment (alexsu52) [inline nncf/onnx/graph/node_utils.py:27]: Please don't forget to add a description.

File: nncf/onnx/graph/transformations/commands_creation.py (hunk 4 above)
Comment (alexsu52) [reply nncf/onnx/graph/transformations/commands_creation.py:1]: `command_creation.py`?

**Review by alexsu52 — COMMENTED** [review]
In general LGTM.

File: nncf/onnx/graph/transformations/commands_creation.py (hunk 4 above)
Comment (andrey-churkin) [reply nncf/onnx/graph/transformations/commands_creation.py]: I renamed to `command_creation.py`.

File: nncf/onnx/graph/node_utils.py (hunk 5 above)
Comment (andrey-churkin) [reply nncf/onnx/graph/node_utils.py]: Done

**Review by KodiaqQ — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]

**Comment by andrey-churkin** [comment]
run onnx pre-commit tests

File: nncf/quantization/algorithms/bias_correction/algorithm.py [hunk 6]
```diff
@@ -264,7 +242,13 @@ def traverse_to_input_layers(node, output):
         for stat_node in stats_nodes:
             input_nodes.update(nncf_graph.traverse_graph(stat_node, traverse_to_input_layers, traverse_forward=False))
 
-        return [input_node.node_name for input_node in input_nodes], [stat_node.node_name for stat_node in stats_nodes]
+        # TODO(andrey-churkin): Do we really need to store it as dict?
```
Comment (kshpv) [inline nncf/quantization/algorithms/bias_correction/algorithm.py:245]: Why can't we resolve this TODO in this PR?

File: nncf/onnx/graph/node_utils.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        with bias (bias is added to the output tensor of that operation),
+        `False` otherwise.
+    """
+    input_tensor_names = node.layer_attributes.input_tensor_names
+    return node.metatype in LAYERS_WITH_BIAS_METATYPES and len(input_tensor_names) > 2
+
+
+def get_bias_value(node: NNCFNode, model: onnx.ModelProto) -> np.ndarray:
+    """
+    Returns the bias value that is applied to the output tensor of the node's operation.
```
Comment (kshpv) [inline nncf/onnx/graph/node_utils.py:39]:
```suggestion
    Returns the bias tensor of the node.
```

File: nncf/onnx/graph/node_utils.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+from nncf.common.graph.graph import NNCFNode
+from nncf.onnx.graph.metatypes.onnx_metatypes import LAYERS_WITH_BIAS_METATYPES
+from nncf.onnx.graph.onnx_graph import ONNXGraph
+from nncf.onnx.graph.metatypes.onnx_metatypes import ONNX_OPERATION_METATYPES
+from nncf.onnx.graph.metatypes.onnx_metatypes import ONNXIdentityMetatype
+
+
+def is_node_with_bias(node: NNCFNode) -> bool:
+    """
+    Checks if a bias is applied to the node's operation or not.
```
Comment (kshpv) [inline nncf/onnx/graph/node_utils.py:26]:
```suggestion
    Checks if a node has bias
```

**Review by kshpv — CHANGES_REQUESTED** [review]

File: nncf/onnx/graph/node_utils.py (hunk 7 above)
Comment (andrey-churkin) [reply nncf/onnx/graph/node_utils.py:39]: Could you please explain what is the reason for this suggestion?

File: nncf/onnx/graph/node_utils.py (hunk 8 above)
Comment (andrey-churkin) [reply nncf/onnx/graph/node_utils.py:26]: Could you please explain what is the reason for this suggestion?

File: nncf/quantization/algorithms/bias_correction/algorithm.py (hunk 6 above)
Comment (andrey-churkin) [reply nncf/quantization/algorithms/bias_correction/algorithm.py]: I removed this todo.

File: nncf/onnx/graph/node_utils.py (hunk 7 above)
Comment (kshpv) [reply nncf/onnx/graph/node_utils.py:39]: Your descriptions confuses. What you mean by applied? Who applies? Does this bias somehow connected to initial node? I suggest to keep it simple

Comment (andrey-churkin) [reply nncf/onnx/graph/node_utils.py]: Done

File: nncf/onnx/graph/node_utils.py (hunk 8 above)
Comment (andrey-churkin) [reply nncf/onnx/graph/node_utils.py]: Done

**Review by kshpv — APPROVED** [review]

**Pull request merged**